### DISCUSSION TITLE
Is there an overview of quest dependencies?

### DISCUSSION THREAD

**Discussion by DerDings** [streetcomplete/StreetComplete#3047] [General]
The sidewalk quest for example requires the way_lit or the maxspeed quest to be answered beforehand on roads other than `highway=residential`

Is there an (graphical) overview of dependencies like this?
If not, does somebody know a tool to create it? Could be either directly from the code or manually by looking at the filters, but it should be easy to modify to keep it up to date. If the tool doesn't require much coding skills, I could start with that.

The sidewalk quest was enabled for trunks recently. However, (with `maxspeed:type` often not given and not gathered by SC when `maxspeed `is present) without a `lit` tag the sidewalk quest will not be asked on a trunk, and the way_lit quest is not (yet) enabled for trunks . I think a dependency overview could help with things like this.

**Comment by matkoniecz** [comment]
I am not aware about any existing tool for this or how upkeep of such graph can be made easy.

Though I am unsure what would be purpose of such tool.

**Reply by peternewman** [reply]
I'd suggest building on the back of my TagInfo WIP PR #2754 might be sensible if someone wanted this. Then the query tag info could be added to the TagInfo JSON too, as well as used as the source data for a graph, but you'd still need to display it nicely somehow. It might be worth looking at something like Doxygen for that, which has class dependencies, although you're really talking about multi-dependencies here, a bit like the class diamond of multi-inheritance.

> Though I am unsure what would be purpose of such tool.

It sounds like @DerDings is suggesting almost for logic validation. For example to see if:

> without a `lit` tag the sidewalk quest will not be asked on a trunk, and the way_lit quest is not (yet) enabled for trunks

Is intentional, or warrants an issue/PR to fix a minor oversight when enabling a tag in one quest but not the related ones.

Although how you render that all nicely is a whole other question.

I guess what's possibly wanted is to show tags which are set by quests and other quests which search on those tags, then you could look a the filter of the upstream tag to see if it matches similar criteria to the downstream one.
